static/app/utils/demoMode/demoTours.tsx

```diff
@@ -0,0 +1,232 @@
+    DemoTourStep.PERFORMANCE_TRANSACTIONS_TABLE,
+    DemoTourStep.PERFORMANCE_SPAN_TREE,
+  ],
+};
+
+const emptyTourState = {
+  currentStepId: undefined,
+  isCompleted: false,
+  isRegistered: true,
+  orderedStepIds: [],
+};
+
+const TOUR_STATE_INITIAL_VALUE: Record<DemoTour, TourState<any>> = {
+  [DemoTour.SIDEBAR]: {
+    ...emptyTourState,
+    orderedStepIds: TOUR_STEPS[DemoTour.SIDEBAR],
+    tourKey: DemoTour.SIDEBAR,
+  },
+  [DemoTour.ISSUES]: {
+    ...emptyTourState,
+    orderedStepIds: TOUR_STEPS[DemoTour.ISSUES],
+    tourKey: DemoTour.ISSUES,
+  },
+  [DemoTour.RELEASES]: {
+    ...emptyTourState,
+    orderedStepIds: TOUR_STEPS[DemoTour.RELEASES],
+    tourKey: DemoTour.RELEASES,
+  },
+  [DemoTour.PERFORMANCE]: {
+    ...emptyTourState,
+    orderedStepIds: TOUR_STEPS[DemoTour.PERFORMANCE],
+    tourKey: DemoTour.PERFORMANCE,
+  },
+};
+
+export function DemoToursProvider({children}: {children: React.ReactNode}) {
+  const org = useOrganization();
+  const [tourState, setTourState] = useLocalStorageState<
+    Record<DemoTour, TourState<any>>
+  >(DEMO_TOURS_STATE_KEY, TOUR_STATE_INITIAL_VALUE);
+
+  const handleStepChange = useCallback(
+    (tourKey: DemoTour, stepId: DemoTourStep) => {
+      setTourState(prev => ({
+        ...prev,
+        [tourKey]: {...prev[tourKey], currentStepId: stepId},
+      }));
+    },
+    [setTourState]
+  );
+
+  const handleEndTour = useCallback(
+    (tourKey: DemoTour) => {
+      setTourState(prev => ({
+        ...prev,
+        [tourKey]: {
+          ...prev[tourKey],
+          currentStepId: undefined,
+          isCompleted: true,
+        },
+      }));
+      recordFinish(tourKey, org.id, org.slug, org);
+    },
+    [setTourState, org]
+  );
+
+  const getTourOptions = useCallback(
+    (tourKey: DemoTour) => ({
+      onEndTour: () => handleEndTour(tourKey),
+      onStepChange: (stepId: DemoTourStep) => handleStepChange(tourKey, stepId),
+      requireAllStepsRegistered: false,
+    }),
+    [handleEndTour, handleStepChange]
+  );
+
+  const sidebarTour = useTourReducer<DemoTourStep>(
+    tourState[DemoTour.SIDEBAR],
+    getTourOptions(DemoTour.SIDEBAR)
+  );
+
+  const issuesTour = useTourReducer<DemoTourStep>(
+    tourState[DemoTour.ISSUES],
+    getTourOptions(DemoTour.ISSUES)
+  );
+
+  const releasesTour = useTourReducer<DemoTourStep>(
+    tourState[DemoTour.RELEASES],
+    getTourOptions(DemoTour.RELEASES)
+  );
+
+  const performanceTour = useTourReducer<DemoTourStep>(
+    tourState[DemoTour.PERFORMANCE],
+    getTourOptions(DemoTour.PERFORMANCE)
+  );
+
+  const tours = useMemo(
+    () => ({
+      [DemoTour.SIDEBAR]: sidebarTour,
+      [DemoTour.ISSUES]: issuesTour,
+      [DemoTour.RELEASES]: releasesTour,
+      [DemoTour.PERFORMANCE]: performanceTour,
+    }),
+    [issuesTour, releasesTour, performanceTour, sidebarTour]
+  );
+
+  return <DemoToursContext.Provider value={tours}>{children}</DemoToursContext.Provider>;
```

Comment:
react 19 lets us use them directly
```suggestion
  return <DemoToursContext value={tours}>{children}</DemoToursContext>;
```